sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/Schema.java

```diff
@@ -816,6 +821,13 @@ public static FieldType iterable(FieldType elementType) {
 
     /** Create a map type for the given key and value types. */
     public static FieldType map(FieldType keyType, FieldType valueType) {
+      if (FieldType.BYTES.equals(keyType)) {
```

Comment:
This is a valid warning, however may cause log spam as the code path is visited on every value.

A Possible solution is to have a private static flag to track if the warning has been emitted so emit this error once.

Reply:
I am not sure that it will cause spam, just one message per one `FieldType` with byte array key in a map.
It's not clear where to have this flag, because `FieldType` object may exist independently of `Schema`, in such case we want to have a warning too. What do you think?

Reply:
yeah you are right, here is create FieldType instead of FieldValue